public/app/core/components/AppChrome/ReturnToPrevious/ReturnToPreviousWrapper.tsx

```diff
@@ -0,0 +1,55 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2, useTheme2 } from '@grafana/ui';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+
+import { ReturnToPrevious } from './ReturnToPrevious';
+
+export const ReturnToPreviousWrapper = () => {
+  const params = useQueryParams()[0];
+  const styles = useStyles2(getStyles);
+  const theme = useTheme2();
+
+  const [paramsExist, setParamsExist] = React.useState(params?.__returnToTitle && params?.__returnToUrl);
+  // Don't show the button whether if there is no params or the URL param matches the current URL
+  const showReturnToPrevious: boolean = paramsExist && location.pathname !== params.__returnToUrl ? true : false;
+
+  // Only show the button on large screens
+  const isLargeScreen = window.innerWidth >= theme.breakpoints.values.xl;
```

Comment:
Does this mean we are not too worried about window resizing? If we want to consider window resizing we can put this in a media query instead?